### PR TITLE
chore: Use same strict tsconfig as in js-ceramic anchor-utils

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: src/merkle/merkle-tree.ts [hunk 1]
```diff
@@ -62,7 +63,7 @@ export class MerkleTree<TData, TLeaf extends TData, TMetadata> {
     /**
      * Tree metadata
      */
-    readonly metadata: TMetadata
+    readonly metadata: TMetadata | null
```
Comment (pawartur) [inline src/merkle/merkle-tree.ts:66]: Can you add a comment about what it means that metadata is null?

File: src/normalize-config.util.ts [hunk 2]
```diff
@@ -29,7 +29,9 @@ export function normalizeConfig(config: Config): void {
  */
 export function cleanupConfigForLogging(config: Config): Record<string, any> {
   const configCopy = cloneDeep(config)
```
Comment (pawartur) [inline src/normalize-config.util.ts:31]: Just a thought, not really a comment to this PR: seeing that we have a function called `cleanupConfigForLogging(...)`, I'd be seriously worried that somebody in future sees that we log configs and logs it themselves, but forgets to put it through the function first.

I guess, I'd prefer to have a custom logging function that we'd use everywhere which would make sure that all the sensitive data is hidden, before putting in logs.

**Review by pawartur — APPROVED** [review]

File: src/services/blockchain/ethereum/ethereum-blockchain-service.ts [hunk 3]
```diff
@@ -66,20 +66,15 @@ async function attempt<T>(
  * @param walletBalance - Available funds.
  */
 function handleInsufficientFundsError(txData: TransactionRequest, walletBalance: BigNumber): void {
-  const txCost = (txData.gasLimit as BigNumber).mul(txData.maxFeePerGas)
+  const txCost = (txData.gasLimit as BigNumber).mul(txData.maxFeePerGas || 1)
```
Comment (pawartur) [inline src/services/blockchain/ethereum/ethereum-blockchain-service.ts:69]: Can you put this `1` as a constant somewhere? Out of curiosity, why `1` is the right number :) ?

Comment (ukstv) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:69]: That's to tell TS it's a safe operation. Going to add a comment there.

Comment (ukstv) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:69]: Or, rather change the code.

File: src/merkle/merkle-tree.ts (hunk 1 above)
Comment (ukstv) [reply src/merkle/merkle-tree.ts:66]: Added: "Tree metadata. Would not be considered for a proof if not available."

File: src/normalize-config.util.ts (hunk 2 above)
Comment (ukstv) [reply src/normalize-config.util.ts:31]: I'd set the probability of anyone misusing logging for this to be close to zero, as it requires pretty deep penetration inside the codebase.

**Pull request merged**